activerecord/lib/active_record/relation.rb

```diff
@@ -507,6 +507,32 @@ def delete_all
       affected
     end
 
+    # Finds and destroys all records matching the specified conditions.
+    # This is short-hand for `relation.where(condition).destroy_all`.
+    # Returns the collection of objects that were destroyed.
+    #
+    # If no record is found, returns empty array.
+    #
+    #   Person.all.destroy_by(name: 'Spartacus', rating: 4)
+    #   Person.all.destroy_by("published_at < ?", 2.weeks.ago)
```

Comment:
This doesn’t seem to match @dhhs description. I don’t think we want `.all` before a destroy_by. It should work like find_by straight on Person.